lib/engine/route.rb

```diff
@@ -59,56 +60,79 @@ def tail
       connection_data[-1]
     end
 
+    def chains
+      connection_data&.map { |c| c[:chain] }
+    end
+
     def connections
```

Comment:
is this method used anymore?

Reply:
Yes, by the UI.

Reply:
Actually, replaced all calls to this with chains